### ISSUE TITLE
Update `no-sparse-arrays` and `no-array-constructor` rules.

### ISSUE THREAD

**Issue by pladaria** [eslint/eslint#14750]
**What rule do you want to change?**
no-sparse-arrays
no-array-constructor

**Does this change cause the rule to produce more or fewer warnings?**
Probably more

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option would be fine

**Please provide some example code that this change will affect:**

1) Rule names are confusing:
* `no-sparse-arrays` allows creating sparse arrays: `Array(123)` is fine for this rule.
* `no-array-constructor` allows using the Array constructor: `Array(123)` is fine for this rule.

2) I don't want to allow creating sparse arrays because it can lead to unwanted behavior, specially when combined with babel's `loose: true` option.

The following code produces different results when babel `loose` option is enabled or not:
```js
[...Array(3)].map((_, i) => console.log(i));
```

Expected result:
```
> [...Array(3)].map((_, i) => console.log(i));
0
1
3
```

With `loose: true` it gets compiled into:

```js
[].concat(Array(123)).map(function (_, i) {
  return console.log(i);
});
```
And that prints nothing ([View in Babel repl](https://babeljs.io/repl/#?browsers=chrome%20%3E%3D%2036&build=&builtIns=usage&corejs=3.6&spec=false&loose=true&code_lz=NoOjEECdIQwTwBQEYBMBmAlAXRAWxgA4IID6ANAAQCWGFAvAHwUDGA9gHYDOrANgKYgerAOYIaGANxA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=true&targets=&version=7.14.7&externalPlugins=babel-plugin-minify-dead-code-elimination%400.5.1%2Cbabel-plugin-minify-guarded-expressions%400.4.4%2Cbabel-plugin-minify-simplify%400.5.1))

So, unless there's an existing rule to avoid this kind of problems, I want to fully disallow creating arrays using `Array(n)` in favor to `Array.from({length: n})`;

**What does the rule currently do for this code?**
https://eslint.org/docs/rules/no-array-constructor
https://eslint.org/docs/rules/no-sparse-arrays

**What will the rule do after it's changed?**
It (optionally) shouldn't allow creating Arrays using `Array(value)` constructor. 

A better alternative would be: `Array.from({length: value})`.

**Are you willing to submit a pull request to implement this change?**
Maybe, I want to get some feedback first and suggestions for the accepted options

**Comment by nzakas** [comment]
As stated in the [docs](https://eslint.org/docs/rules/no-sparse-arrays), no-sparse-arrays is intended to catch array literals with empty slots, so we won’t be changing that rule. If you want to avoid the use of `Array(123)`, you can use no-restricted-syntax or create a custom rule. 

And the [docs](https://eslint.org/docs/rules/no-array-constructor) for no-array-constructor explicitly states:

> The exception is when the Array constructor is used to intentionally create sparse arrays of a specified size by giving the constructor a single numeric argument.

The rule was intentionally created to allow the pattern you want to disallow.

We aren’t making any further changes to rules that enforce stylistic preferences like these. Your best bet is to create your own rule that does exactly what you want or else use no-restricted-syntax. 

in the future, please open one issue per change you want. It makes our lives easier to consider just one change at a time. Thanks.
